### PR TITLE
feat: passing the inputs values using `difyChatbotConfig`

### LINKED ISSUE #5575: Pass input parameters to iframe
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Same request as [this closed issue](https://github.com/langgenius/dify/issues/2841), there seems to be no simple way to pass data to user input fields when embedding an iframe. The current implementation has poor user experience as the user must type in some text which could be easily pre-filled.

### 2. Additional context or comments

Even better than pre-filling inputs would be to pre-fill inputs _and_ start the chat if all required inputs are present.

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by yoyocircle**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

> Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. Close issue syntax: `Fixes #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

- Adding the key `inputs` to `difyChatbotConfig` and make it fill the inputs when initializing the chatbot.
- Add gzip compression and decompression to make the inputs could be longer. Since the url length limitation depends on the web server configuration, I decide to pass it anyway and output an error log if it's longer than 2048.

related issues: Close #5575

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Some test values:
| original value | processed |
| - | - |
| test | H4sIADTql2YA%2FwXAMQ0AAADCMLUooPaPxS4Mfn%2FYBAAAAA%3D%3D |
| 1 | H4sIAFSUmGYA%2FwXAgQgAAAAAIKb95U6379yDAQAAAA%3D%3D |

- [x] Test chatbot: by setting up a new bot with inputs. and the access the chatbot with parameters. 
- [x] setting text-input to any texts 
- [x] setting paragraph to any texts
- [x] setting number with a number
- [x] setting option with a expected value
- [x] validate unexpected value/option for number/option
- [x] truncate the string for text-input and paragraph if the max length is set

### CODE REVIEW

**Comment by crazywoola** [comment]
Please notice it will conflict with https://github.com/langgenius/dify/pull/6360 on  `embed.min.js`. It needs to be rebuilt again.

**Comment by yoyocircle** [comment]
> Please notice it will conflict with #6360 on `embed.min.js`. It needs to be rebuilt again.

Hello @crazywoola, I will try to solve the conflict after #6360 is merged into main.

**Comment by crazywoola** [comment]
Merged, please fix the lint errors as well.

**Comment by yoyocircle** [comment]
Hi @crazywoola, the changes should be done, and also add another feature to compress the strings.

File: web/app/components/base/chat/embedded-chatbot/hooks.tsx [hunk 1]
```diff
@@ -140,6 +144,25 @@ export const useEmbeddedChatbot = () => {
       }
     })
   }, [appParams])
+  useEffect(async () => {
+    async function decodeBase64AndDecompress(base64String) {
```
Comment (crazywoola) [inline web/app/components/base/chat/embedded-chatbot/hooks.tsx]: Suggest to move this util function to `web/app/components/base/chat/utils`

**Review by crazywoola — CHANGES_REQUESTED** [review]

**Comment by yoyocircle** [comment]
Hi @crazywoola, the functions were moved. And also validate the values when the inputs type is number or select.

**Comment by yoyocircle** [comment]
> Can you add my wechat: crazyphage

sure, will add you tonight.

File: web/app/components/base/chat/utils.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+async function decodeBase64AndDecompress(base64String) {
```
Comment (crazywoola) [inline web/app/components/base/chat/utils.ts]:
```suggestion
async function decodeBase64AndDecompress(base64String: string) {
```

**Review by crazywoola — COMMENTED** [review]
text-input is not working, the rest of types are working for me.

**Comment by yoyocircle** [comment]
> text-input is not working, the rest of types are working for me.

This is fixed. 
And added the length check for text-input and paragraph.

**Review by crazywoola — APPROVED** [review]

**Comment by xbduck** [comment]
inputs: {
        // 键是变量名
        // 例如：
        name: "NAME",
      },
配置的name对应的默认值 页面上没有读取到

**Comment by Amandine95** [comment]
is the key in inputs same with the params  name set in start node ?

**Comment by Amandine95** [comment]
> inputs: { // 键是变量名 // 例如： name: "NAME", }, 配置的name对应的默认值 页面上没有读取到


问题结局了吗？

**Comment by xbduck** [comment]
自动回复：您好，你的邮件已经收到，我会尽快拜读，祝你天天开心，万事如意！

**Comment by biglazybug** [comment]
> > inputs: { // 键是变量名 // 例如： name: "NAME", }, 配置的name对应的默认值 页面上没有读取到
> 
> 问题如何解决的？

问题解决了吗

**Comment by xbduck** [comment]
自动回复：您好，你的邮件已经收到，我会尽快拜读，祝你天天开心，万事如意！

**Comment by wumingyu12** [comment]
具体在工作流中怎么用这个input 里面的变量的

**Comment by xbduck** [comment]
自动回复：您好，你的邮件已经收到，我会尽快拜读，祝你天天开心，万事如意！

**Pull request merged**